### PR TITLE
initial BubbleSort code & test

### PR DESCRIPTION
**Pull request by ianwer**

### CODE REVIEW

File: src/main/java/com/start/code/BubbleSort.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                                 arr[j-1] = arr[j];  
+                                 arr[j] = temp;  
+                         }  
+                          
+                 }  
+         }  
+  
+    } 
+	
+	static void print(int[] arr){
```
Comment (toliaqat) [inline src/main/java/com/start/code/BubbleSort.java]: Remove this method. Not needed anymore.

File: src/main/java/com/start/code/BubbleSort.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.start.code;
+
+public class BubbleSort {
+	static void SortMyArray(int[] arr) {  
+        int n = arr.length;  
+        int temp = 0;  
+         for(int i=0; i < n; i++){
```
Comment (toliaqat) [inline src/main/java/com/start/code/BubbleSort.java:7]: Need to add spaces around operators.

File: src/main/java/com/start/code/BubbleSort.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                          
+                 }  
+         }  
+  
+    } 
+	
+	static void print(int[] arr){
+		for(int i=0; i < arr.length; i++){  
+            System.out.print(arr[i] + " ");
+        }
```
Comment (toliaqat) [inline src/main/java/com/start/code/BubbleSort.java]: Please remove this method. Not needed anymore.

File: src/test/java/com/start/code/TestBubbleSort.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.start.code;
+
+import com.start.code.BubbleSort;
+
+import junit.framework.TestCase;
+
+public class TestBubbleSort extends TestCase {
+	
+	public void testBubbleSort() {
```
Comment (toliaqat) [inline src/test/java/com/start/code/TestBubbleSort.java]: Rename this test method, to state what is being tested in this particular test.
And add few more tests.

File: src/main/java/com/start/code/BubbleSort.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package com.start.code;
+
+public class BubbleSort {
+	static void SortMyArray(int[] arr) {
```
Comment (toliaqat) [inline src/main/java/com/start/code/BubbleSort.java:4]: Rename to Sort

Comment (toliaqat) [inline src/main/java/com/start/code/BubbleSort.java:4]: Check that arr is not null.

File: src/test/java/com/start/code/TestBubbleSort.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+public class TestBubbleSort extends TestCase {
+	
+	public void testBubbleSort() { 
+        int arr[] ={2,1};
+        BubbleSort.SortMyArray(arr);//Sorting
+        for(int i = 0; i < arr.length-1; i++){
+        	assertTrue(arr[i] <= arr[i+1]);
+        }
+    }  
+}
```
Comment (toliaqat) [inline src/test/java/com/start/code/TestBubbleSort.java]: Add test to check for null.
Add test to check behavior on empty list.
Etc.

File: src/main/java/com/start/code/BubbleSort.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package com.start.code;
+
+public class BubbleSort {
```
Comment (toliaqat) [inline src/main/java/com/start/code/BubbleSort.java:3]: Add methods to sort other number types (i.e. double, long), and String array and a List.

**Comment by ianwer** [comment]
I removed the method saved it, but I was trying to run test and clicked on new and junit test case. It asked for the name and entered "BubbleSort", it is showing many errors. 
I stopped on the same stage, let me know how to fix it?
Regards,Iqbal Anwer 

    On Thursday, March 30, 2017 9:40 AM, Touseef Liaqat <notifications@github.com> wrote:
 

 @toliaqat commented on this pull request.In src/main/java/com/start/code/BubbleSort.java:> +        int temp = 0;  
+         for(int i=0; i < n; i++){  
+                 for(int j=1; j < (n-i); j++){  
+                          if(arr[j-1] > arr[j]){  
+                                 //swap elements  
+                                 temp = arr[j-1];  
+                                 arr[j-1] = arr[j];  
+                                 arr[j] = temp;  
+                         }  
+                          
+                 }  
+         }  
+  
+    } 
+	
+	static void print(int[] arr){
Remove this method. Not needed anymore.—

**Comment by toliaqat** [comment]
Following command will clean up your repository if you have not committed bad code.

```
git clean -x -d -f
```

After running above command, make sure your tests are running by running following command.

```
mvn test
```

After above returns successfully. Do the refactoring of code. You do not need to any new FILE or CLASS. So this problem should not come.

Next time, I want to report this kind of problem with following information with clear steps to repro.

Output of `git status`
Output of `mvn test`

Thanks,

**Comment by ianwer** [comment]
C:\Users\Iqbal\code>git clean -x -d -fRemoving .classpathRemoving .projectRemoving .settings/Removing commands.txtRemoving src/test/java/com/start/code/BubbleSort.javaRemoving target/
C:\Users\Iqbal\code>git clean -x -d -fRemoving .project
C:\Users\Iqbal\code>mvn test[INFO] Scanning for projects...[INFO][INFO] ------------------------------------------------------------------------[INFO] Building code 1.0-SNAPSHOT[INFO] ------------------------------------------------------------------------[INFO][INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ code ---[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent![INFO] skip non existing resourceDirectory C:\Users\Iqbal\code\src\main\resources[INFO][INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ code ---[INFO] Changes detected - recompiling the module![WARNING] File encoding has not been set, using platform encoding Cp1252, i.e. build is platform dependent![INFO] Compiling 2 source files to C:\Users\Iqbal\code\target\classes[INFO][INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ code ---[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent![INFO] skip non existing resourceDirectory C:\Users\Iqbal\code\src\test\resources[INFO][INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ code ---[INFO] Changes detected - recompiling the module![WARNING] File encoding has not been set, using platform encoding Cp1252, i.e. build is platform dependent![INFO] Compiling 2 source files to C:\Users\Iqbal\code\target\test-classes[INFO][INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ code ---[INFO] Surefire report directory: C:\Users\Iqbal\code\target\surefire-reports
------------------------------------------------------- T E S T S-------------------------------------------------------Running com.start.code.MainTestTests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 secRunning com.start.code.TestBubbleSortTests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
Results :
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
[INFO] ------------------------------------------------------------------------[INFO] BUILD SUCCESS[INFO] ------------------------------------------------------------------------[INFO] Total time: 9.123 s[INFO] Finished at: 2017-03-31T11:53:31-07:00[INFO] Final Memory: 14M/106M[INFO] ------------------------------------------------------------------------
C:\Users\Iqbal\code>git statusOn branch masterYour branch is up-to-date with 'origin/master'.Changes not staged for commit:  (use "git add <file>..." to update what will be committed)  (use "git checkout -- <file>..." to discard changes in working directory)
        modified:   src/main/java/com/start/code/BubbleSort.java
no changes added to commit (use "git add" and/or "git commit -a")
C:\Users\Iqbal\code>
 

    On Friday, March 31, 2017 11:29 AM, Touseef Liaqat <notifications@github.com> wrote:
 

 Following command will clean up your repository if you have not committed bad code.git clean -x -d -f
After running above command, make sure your tests are running by running following command.mvn test
After above returns successfully. Do the refactoring of code. You do not need to any new FILE or CLASS. So this problem should not come.Next time, I want to report this kind of problem with following information with clear steps to repro.Output of git status
Output of mvn testThanks,—

**Comment by toliaqat** [comment]
So it seems like you were able to recover. Now back to fixing the issues I raised in code review.
Also for next time, please use code block to paste any logs from the command shell. Without that it is hard to read them. Use can enclose your logs between
 \`\`\` 
\<your code or output log\>
 \`\`\`

Following is the same log you reported earlier in readable form.

```
 C:\Users\Iqbal\code>git clean -x -d -f
Removing .classpath
Removing .projectRemoving .settings/
Removing commands.txt
Removing src/test/java/com/start/code/BubbleSort.java
Removing target/

C:\Users\Iqbal\code>mvn test
[INFO] Scanning for projects...[INFO][INFO] ------------------------------------------------------------------------[INFO] Building code 1.0-SNAPSHOT[INFO] ------------------------------------------------------------------------[INFO][INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ code ---[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent![INFO] skip non existing resourceDirectory C:\Users\Iqbal\code\src\main\resources[INFO][INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ code ---[INFO] Changes detected - recompiling the module![WARNING] File encoding has not been set, using platform encoding Cp1252, i.e. build is platform dependent![INFO] Compiling 2 source files to C:\Users\Iqbal\code\target\classes[INFO][INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ code ---[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent![INFO] skip non existing resourceDirectory C:\Users\Iqbal\code\src\test\resources[INFO][INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ code ---[INFO] Changes detected - recompiling the module![WARNING] File encoding has not been set, using platform encoding Cp1252, i.e. build is platform dependent![INFO] Compiling 2 source files to C:\Users\Iqbal\code\target\test-classes[INFO][INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ code ---[INFO] Surefire report directory: C:\Users\Iqbal\code\target\surefire-reports
------------------------------------------------------- T E S T S-------------------------------------------------------Running com.start.code.MainTestTests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 secRunning com.start.code.TestBubbleSortTests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
Results :
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 9.123 s[INFO] Finished at: 2017-03-31T11:53:31-07:00[INFO] Final Memory: 14M/106M
[INFO] ------------------------------------------------------------------------
C:\Users\Iqbal\code>git status
On branch masterYour branch is up-to-date with 'origin/master'.Changes not staged for commit:  (use "git add <file>..." to update what will be committed)  (use "git checkout -- <file>..." to discard changes in working directory)
        modified:   src/main/java/com/start/code/BubbleSort.java
no changes added to commit (use "git add" and/or "git commit -a")
C:\Users\Iqbal\code>
```

File: src/test/java/com/start/code/TestBubbleSort.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+	
+	public void test0BubbleSort() { 
+        int arr[] ={2,1};
+        BubbleSort.SortMyArray(arr);//Sorting
+        for(int i = 0; i < arr.length-1; i++){
+        	assertTrue(arr[i] <= arr[i+1]);
+        }
+    } 
+	
+	public void test1BubbleSort() {
```
Comment (toliaqat) [inline src/test/java/com/start/code/TestBubbleSort.java]: Use camelCase for method names. And avoid number in them. This method could be called, testAlreadySortedList()

File: src/test/java/com/start/code/TestBubbleSort.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        BubbleSort.SortMyArray(arr);//Sorting
+        for(int i = 0; i < arr.length-1; i++){
+        	assertTrue(arr[i] <= arr[i+1]);
+        }
+    }
+	
+	public void test2BubbleSort() { 
+        int arr[] ={};
+        BubbleSort.SortMyArray(arr);//Sorting
+        for(int i = 0; i < arr.length-1; i++){
```
Comment (toliaqat) [inline src/test/java/com/start/code/TestBubbleSort.java]: This loop is not required. You just need to verify the array size.

**Comment by toliaqat** [comment]
+2
